### ISSUE TITLE
Create class for HTTP request

### ISSUE THREAD

**Issue by 0xOnyx** [0xOnyx/42_webserv#6]
make a class that allows you to generate HTTP requests and that returns all of these in a std::string

**Comment by LeRoiErrant** [comment]
By HTTP request you mean HTTP response ?

**Comment by LeRoiErrant** [comment]
Accordingly to what we discussed before, by HTTP request you mean HTTP response I think ?

**Comment by 0xOnyx** [comment]
Yes, we should create a template class that would allow creating requests according to the HTTP1.1 standard. Here's what it should take as parameters for the constructor:

The request status, e.g., 200, 404, etc., as an integer
The length of the body as a size_t
The type of the body as a std::string
If the length is greater than 0, then take the body as a std::string.
Then, you should return a correctly formatted std::string that I can send when the socket is available for writing, as follows:

HTTP/1.1 200 OK
Content-Type: text/html

<!DOCTYPE html>
<html>
<head>
    <title>Hello World</title>
</head>
<body>
    <h1>Hello World!</h1>
</body>
</html>
